src/main/java/io/supertokens/jwt/JWTSigningKey.java

```diff
@@ -0,0 +1,247 @@
+import io.supertokens.exceptions.QuitProgramException;
+import io.supertokens.jwt.exceptions.UnsupportedJWTSigningAlgorithmException;
+import io.supertokens.pluginInterface.STORAGE_TYPE;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.pluginInterface.jwt.JWTAsymmetricSigningKeyInfo;
+import io.supertokens.pluginInterface.jwt.JWTSymmetricSigningKeyInfo;
+import io.supertokens.pluginInterface.jwt.JWTSigningKeyInfo;
+import io.supertokens.pluginInterface.jwt.exceptions.DuplicateKeyIdException;
+import io.supertokens.pluginInterface.session.SessionStorage;
+import io.supertokens.pluginInterface.session.noSqlStorage.SessionNoSQLStorage_1;
+import io.supertokens.pluginInterface.session.sqlStorage.SessionSQLStorage;
+import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.utils.Utils;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JWTSigningKey extends ResourceDistributor.SingletonResource {
+    public static final String RESOURCE_KEY = "io.supertokens.jwt.JWTSigningKey";
+    private final Main main;
+
+    public enum SupportedAlgorithms {
+        RS256;
+
+        public String getAlgorithmType() {
+            if (this == SupportedAlgorithms.RS256) {
+                return "rsa";
+            }
+
+            return "";
+        }
+    }
+
+    private JWTSigningKey(Main main) {
+        this.main = main;
+    }
+
+    public static JWTSigningKey getInstance(Main main) {
+        JWTSigningKey instance = (JWTSigningKey) main.getResourceDistributor().getResource(RESOURCE_KEY);
+
+        if (instance == null) {
+            instance = new JWTSigningKey(main);
+            main.getResourceDistributor().setResource(RESOURCE_KEY, instance);
+        }
+
+        return instance;
+    }
+
+    /**
+     * Used to get all signing keys (symmetric or asymmetric) from storage
+     * @return List of {@link JWTSigningKeyInfo}. Asymmetric keys use {@link JWTAsymmetricSigningKeyInfo} and symmetric keys use {@link JWTSymmetricSigningKeyInfo}
+     * @throws StorageQueryException If there is an error interacting with the database
+     * @throws StorageTransactionLogicException If there is an error interacting with the database
+     */
+    public List<JWTSigningKeyInfo> getAllSigningKeys()
+            throws StorageQueryException, StorageTransactionLogicException {
+        SessionStorage storage = StorageLayer.getSessionStorage(main);
+
+        if (storage.getType() == STORAGE_TYPE.SQL) {
+            SessionSQLStorage sqlStorage = (SessionSQLStorage) storage;
+
+            return sqlStorage.startTransaction(con -> {
+               List<JWTSigningKeyInfo> keys;
+               List<JWTSigningKeyInfo> keysFromStorage = sqlStorage.getJWTSigningKeys_Transaction(con);
+
+               if (keysFromStorage == null) {
```

Comment:
When would `keysFromStorage` be `null`? If there are no keys, shouldn't the function return an empty `List`?

Reply:
So my thinking was that from the storage layer we should return a list if there are keys, or null otherwise. The storage layer should not assume that the caller does not want to handle nulls. So instead of the storage layer returning an empty List this method handles null.

Reply:
I disagree. I think the storage layer should return an empty array. It keeps the types clean (the fewer variations, the better). You should make sure that the storage layer returns an empty list in this case.

Reply:
So thinking about it a little more I think you're right, ill make the change

Reply:
Done